ImageProcessing/OpeningPlacement.py

```diff
@@ -19,51 +21,59 @@ def __init__(self):
     def process(self, parent):
         """smooth the data"""
         res = []
+        blank = np.zeros((parent.height, parent.width), np.uint8)
         for i, room in enumerate(self.img):
-            blank = np.zeros((parent.height, parent.width), np.uint8)
             render_room(blank, room, gray=True)
-
             new_room = copy.deepcopy(room)
             for opening in self.desc.openings:
-                shift = 0
-                if opening._type == 'door':
-                    shift = OpeningPlacement.WIDER
-
-                p1 = opening.placement.point_1
-                p2 = opening.placement.point_2
-                _sx = int(min(p1.x, p2.x)) - shift
-                _ex = int(max(p1.x, p2.x)) + shift
-
-                _sy = int(min(p1.y, p2.y)) - shift
-                _ey = int(max(p1.y, p2.y)) + shift
-
-                rect_img = blank[_sy:_ey, _sx:_ex]
-
-                maskRect = np.where(rect_img != 0)
-                maskPts = np.hstack((maskRect[1][:, np.newaxis], maskRect[0][:, np.newaxis]))
-                sx, sy, ex, ey = parent.width, parent.height, 0, 0
-                if len(maskPts):
-                    for _ in maskPts:
-                        sx = min(sx, _[0])
-                        sy = min(sy, _[1])
-                        ex = max(ex, _[0])
-                        ey = max(ey, _[1])
-
-                    sy += p1.y - shift
-                    sx += p1.x - shift
-                    ex += p1.x - shift
-                    ey += p1.y - shift
+                new_item = self.find_opening_place(new_room.walls, opening)
+                if new_item:
+                    new_room.openings.append(new_item)
+                render_room(blank, new_room)
```

Comment:
Blank and render_room are ow useless cause you are finding geometrical intersections